### PR TITLE
Add MapPatch API to endpointRouteBuilderExtensions

### LINKED ISSUE #36198: Add MapPatch overloads to routing
## Background and Motivation

Today we have shortcuts for the most common HTTP methods for route handling (GET, POST, PUT, DELETE) but we don't have PATCH (or OPTIONS, TRACE, HEAD). We should add MapPatch to the set of methods we support. MapPatch is also interesting because people can now write custom binders in minimal APIs to handle the body. Today calling MapMethods(..) is the workaround (https://github.com/felipeoriani/MinimalBlogApi/blob/83732f1785a4c01a63ccf6909ca2a8190c63b4c9/Program.cs#L118) but it's kinda gross.

## Proposed API

```diff
namespace Microsoft.AspNetCore.Builder
{
    public static class DelegateEndpointRouteBuilderExtensions
    {
+       public static DelegateEndpointConventionBuilder MapPatch(this IEndpointRouteBuilder endpoints, string pattern, Delegate handler);
    }
}
```

## Usage Examples

``` C#
app.MapPatch("/patch", (Patch<Model> patch)
{
      return patch.Changes;
});
```

## Risks

Low. 

Questions: 
- Do we also add RequestDelegate overloads here?

### PR DESCRIPTION
**Pull request by noamyogev84**
- [ x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [ x] You've included unit or integration tests for your change, where applicable.
- [x ] You've included inline docs for your change, where applicable.
- [ x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

**Add MapPatch API to endpointRouteBuilderExtensions**
just add the patch verb with the same behavior as other verbs in this file

**PR Description**
As written above
Still have some issues building this project, not sure that i find `DelegateEndpointConventionBuilder` as in the original suggestion

Fixes #36198 (in this specific format)

### CODE REVIEW

**Review by captainsafia — COMMENTED** [review]
Thanks for opening this PR! We utilize the PublicAPI analyzer to track API changes in our repo, so you'll need to update the nearest `PublicAPI.Unshipped.txt` file adjacent to the new API.

We'll also need to include some tests for this PR. The best place to add them would be in https://github.com/dotnet/aspnetcore/blob/2b0e81f2ff05b6cc9378850d86637ea964e07723/src/Http/Routing/test/UnitTests/Builder/RouteHandlerEndpointRouteBuilderExtensionsTest.cs

**Comment by noamyogev84** [comment]
@captainsafia 
sure, on it ✨

**Comment by noamyogev84** [comment]
Hi @captainsafia what's up
ready for review

File: src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs [hunk 1]
```diff
@@ -88,6 +88,22 @@ public static class EndpointRouteBuilderExtensions
             return MapMethods(endpoints, pattern, DeleteVerb, requestDelegate);
         }
 
+        /// <summary>
+        /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that matches HTTP PATCH requests
+        /// for the specified pattern.
+        /// </summary>
+        /// <param name="endpoints"></param>
```
Comment (captainsafia) [inline src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs:95]: Nit: these params need to be filled in.

File: src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs [hunk 2]
```diff
@@ -255,6 +271,23 @@ public static class EndpointRouteBuilderExtensions
...
         }
 
+
+        /// <summary>
+        /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that matches HTTP PATCH requests
+        /// for the specified pattern.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="handler">The delegate executed when the endpoint is matched.</param>
```
Comment (captainsafia) [inline src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs:281]:
```suggestion
        /// <param name="handler">The <see cref="Delegate" /> executed when the endpoint is matched.</param>
```

**Review by captainsafia — APPROVED** [review]
Looks good overall! Left a few comments on docs that need to be addressed on docs but no functional/test related issues.

File: src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs (hunk 1 above)
Comment (noamyogev84) [reply src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs]: Added comments.
Thanks @captainsafia 🐱‍🏍

**Review by captainsafia — APPROVED** [review]

**Pull request merged**